src/mount/mount-tool.c

```diff
@@ -363,13 +363,11 @@ static int parse_argv(int argc, char *argv[]) {
                         if (arg_transport == BUS_TRANSPORT_LOCAL) {
                                 _cleanup_free_ char *p = NULL;
 
-                                r = path_make_absolute_cwd(argv[optind+1], &p);
+                                r = path_make_absolute_cwd(argv[optind+1], &arg_mount_where);
                                 if (r < 0)
                                         return log_error_errno(r, "Failed to make path %s absolute: %m", argv[optind+1]);
 
-                                arg_mount_where = canonicalize_file_name(p);
-                                if (!arg_mount_where)
-                                        return log_error_errno(errno, "Failed to canonicalize path %s: %m", p);
+                                path_kill_slashes(arg_mount_where);
```

Comment:
This doesn't look right.

canonicalize_file_name() resolves symlinks and removes "." and "..", and that's something we should do really, as mount unit names and such cannot contain symlinks or . and .. , and should reference the normalized version of that path.

We should probably rework this to use chase_symlinks() with CHASE_NONEXISTANT, so that we are fine with non-existing paths